### PR TITLE
Add ability to sweep wallet

### LINKED ISSUE #3265: Sweep all funds from GUI
Allow for an easy way to send all funds out of the wallet from GUI

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #3265 

Adds a MAX button that will change it to sweep the wallet

image

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala [hunk 1]
```diff
@@ -278,6 +278,20 @@ trait WalletApi extends StartStopAsync[WalletApi] {
...
+
+  def sweepWallet(address: BitcoinAddress, feeRateOpt: Option[FeeUnit])(implicit
+      ec: ExecutionContext): Future[Transaction] = {
+    for {
+      feeRate <- determineFeeRate(feeRateOpt)
+      tx <- sweepWallet(address, feeRate)
+    } yield tx
+  }
+
+  def sweepWallet(address: BitcoinAddress, feeRate: FeeUnit)(implicit
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala:295]: So.... what is the difference between `emptyWallet()` and `sweepWallet()` ? Neither of them have scaladocs on them

File: wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala [hunk 2]
```diff
@@ -545,6 +545,14 @@ abstract class Wallet
     } yield tx
   }
 
+  override def sweepWallet(address: BitcoinAddress, feeRate: FeeUnit)(implicit
+      ec: ExecutionContext): Future[Transaction] = {
+    for {
+      outpoints <- listUtxos().map(_.map(_.outPoint))
+      tx <- sendFromOutPoints(outpoints, address, feeRate)
+    } yield tx
```
Comment (Christewart) [inline wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala:557]: nit: Maybe worth adding a `logger.info(s"Sweeping wallet balance=${balance} to address=${address}")`

**Review by Christewart — CHANGES_REQUESTED** [review]
A unit test would be good too.

File: core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala (hunk 1 above)
Comment (benthecarman) [reply core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala:295]: Oh was the same thing, removed `emptyWallet` as we already have `isEmpty` so could be confusing

**Review by Christewart — APPROVED** [review]

**Pull request merged**